### PR TITLE
[functions list] Add insertion and extraction functions (#94)

### PR DESCRIPTION
**Pull request by rjiejie**

### CODE REVIEW

File: intrinsic_funcs/12_miscellaneous_vector_functions.md [hunk 1]
```diff
@@ -568,4 +568,80 @@ vfloat64m1_t vundefined_f64m1 ();
...
+vuint64m8_t vset_u64m8 (vuint64m8_t dest, size_t index, vuint64m1_t val);
+vfloat16m2_t vset_f16m2 (vfloat16m2_t dest, size_t index, vfloat16m1_t val);
+vfloat16m4_t vset_f16m4 (vfloat16m4_t dest, size_t index, vfloat16m1_t val);
+vfloat16m8_t vset_f16m8 (vfloat16m8_t dest, size_t index, vfloat16m1_t val);
+vfloat32m2_t vset_f32m2 (vfloat32m2_t dest, size_t index, vfloat32m1_t val);
+vfloat32m4_t vset_f32m4 (vfloat32m4_t dest, size_t index, vfloat32m1_t val);
+vfloat32m8_t vset_f32m8 (vfloat32m8_t dest, size_t index, vfloat32m1_t val);
+vfloat64m2_t vset_f64m2 (vfloat64m2_t dest, size_t index, vfloat64m1_t val);
+vfloat64m4_t vset_f64m4 (vfloat64m4_t dest, size_t index, vfloat64m1_t val);
+vfloat64m8_t vset_f64m8 (vfloat64m8_t dest, size_t index, vfloat64m1_t val);
```
Comment (topperc) [inline intrinsic_funcs/12_miscellaneous_vector_functions.md:608]: What about insert m2 into m4/m8, or m4 into m8?

File: intrinsic_funcs/12_miscellaneous_vector_functions.md [hunk 2]
```diff
@@ -568,4 +568,80 @@ vfloat64m1_t vundefined_f64m1 ();
...
+vuint64m1_t vget_u64m8_u64m1 (vuint64m8_t src, size_t index);
+vfloat16m1_t vget_f16m2_f16m1 (vfloat16m2_t src, size_t index);
+vfloat16m1_t vget_f16m4_f16m1 (vfloat16m4_t src, size_t index);
+vfloat16m1_t vget_f16m8_f16m1 (vfloat16m8_t src, size_t index);
+vfloat32m1_t vget_f32m2_f32m1 (vfloat32m2_t src, size_t index);
+vfloat32m1_t vget_f32m4_f32m1 (vfloat32m4_t src, size_t index);
+vfloat32m1_t vget_f32m8_f32m1 (vfloat32m8_t src, size_t index);
+vfloat64m1_t vget_f64m2_f64m1 (vfloat64m2_t src, size_t index);
+vfloat64m1_t vget_f64m4_f64m1 (vfloat64m4_t src, size_t index);
+vfloat64m1_t vget_f64m8_f64m1 (vfloat64m8_t src, size_t index);
```
Comment (topperc) [inline intrinsic_funcs/12_miscellaneous_vector_functions.md:646]: m2 from m4/m8 and m4 from m8?

File: intrinsic_funcs/12_miscellaneous_vector_functions.md (hunk 1 above)
Comment (rjiejie) [reply intrinsic_funcs/12_miscellaneous_vector_functions.md]: Fixed.

File: intrinsic_funcs/12_miscellaneous_vector_functions.md (hunk 2 above)
Comment (rjiejie) [reply intrinsic_funcs/12_miscellaneous_vector_functions.md:710]: Fixed.

**Comment by rjiejie** [comment]
@topperc Any comment for new commit ?

**Comment by topperc** [comment]
I think this looks good to me. @kito-cheng what do you think? Do any other files need to be updated?

**Comment by kito-cheng** [comment]
LGTM.

**Comment by kito-cheng** [comment]
We could add that into our internal generator and then update other file later.

File: intrinsic_funcs/12_miscellaneous_vector_functions.md [hunk 3]
```diff
@@ -568,4 +568,146 @@ vfloat64m1_t vundefined_f64m1 ();
...
+vfloat32m4_t vset_f32m4_f32m2 (vfloat32m4_t dest, size_t index, vfloat32m2_t val);
+vfloat32m8_t vset_f32m8_f32m1 (vfloat32m8_t dest, size_t index, vfloat32m1_t val);
+vfloat32m8_t vset_f32m8_f32m2 (vfloat32m8_t dest, size_t index, vfloat32m2_t val);
+vfloat32m8_t vset_f32m8_f32m4 (vfloat32m8_t dest, size_t index, vfloat32m4_t val);
+vfloat64m2_t vset_f64m2_f64m1 (vfloat64m2_t dest, size_t index, vfloat64m1_t val);
+vfloat64m4_t vset_f64m4_f64m1 (vfloat64m4_t dest, size_t index, vfloat64m1_t val);
+vfloat64m4_t vset_f64m4_f64m2 (vfloat64m4_t dest, size_t index, vfloat64m2_t val);
+vfloat64m8_t vset_f64m8_f64m1 (vfloat64m8_t dest, size_t index, vfloat64m1_t val);
+vfloat64m8_t vset_f64m8_f64m2 (vfloat64m8_t dest, size_t index, vfloat64m2_t val);
+vfloat64m8_t vset_f64m8_f64m4 (vfloat64m8_t dest, size_t index, vfloat64m4_t val);
```
Comment (kito-cheng) [inline intrinsic_funcs/12_miscellaneous_vector_functions.md:641]: Naming of intrinsic must end with return type.

So
`vfloat64m8_t vset_f64m8_f64m4 (vfloat64m8_t dest, size_t index, vfloat64m4_t val);`
should be
`vfloat64m8_t vset_f64m4_f64m8 (vfloat64m8_t dest, size_t index, vfloat64m4_t val);`

See also: https://github.com/riscv/rvv-intrinsic-doc/blob/master/rvv-intrinsic-rfc.md#naming-rules

**Review by kito-cheng — COMMENTED** [review]
I would prefer add `_v` to the middle of function name for more consistent naming with other functions like `vlmul_trunc_v_*`, `vlmul_ext_v_*` and `vreinterpret_v_*`.

File: intrinsic_funcs/12_miscellaneous_vector_functions.md (hunk 3 above)
Comment (rjiejie) [reply intrinsic_funcs/12_miscellaneous_vector_functions.md]: Fixed.

**Comment by kito-cheng** [comment]
Could you also add description  to `rvv-intrinsic-rfc.md`?

https://github.com/riscv/rvv-intrinsic-doc/blob/master/rvv-intrinsic-rfc.md#lmul-truncation-and-lmul-extension-functions

**Comment by rjiejie** [comment]
> Could you also add description to `rvv-intrinsic-rfc.md`?
> 
> https://github.com/riscv/rvv-intrinsic-doc/blob/master/rvv-intrinsic-rfc.md#lmul-truncation-and-lmul-extension-functions

Fixed.

**Comment by kito-cheng** [comment]
Thanks :)

**Pull request merged**